src/features/rewards/tableAdsHistory/index.tsx

```diff
@@ -0,0 +1,240 @@
+          {
+            this.getRows(rows)
+          }
+        </tbody>
+      </StyledTable>
+    )
+  }
+
+  getHeader = (header: string[]) => {
+    if (!header) {
+      return
+    }
+
+    return (
+      <>
+        <StyledTHFirst>{header[0]}</StyledTHFirst>
+        <StyledTHOther>{header[1]}</StyledTHOther>
+        <StyledTHLast>{header[2]}</StyledTHLast>
+      </>
+    )
+  }
+
+  getRows = (rows: DetailRow[]) => {
+
+    return rows.map((row: DetailRow, i: number) => {
+      const detailRows = this.getDetailRows(row.adDetailRows)
+      return (
+          <AdRowsDetails key={i} row={row} rowIndex={i} detailRows={detailRows} />
+      )
+    })
+  }
+
+  getCategoryContent = (content: CategoryContent) => {
+    return (
+      <StyledCategoryContentDiv>
+        <StyledCategoryName>
+          {content.category}
+        </StyledCategoryName>
+        <StyledCategoryActions>
+          <CategoryLikePicker
+            optAction={content.optAction}
+            onOptIn={content.onOptInAction}
+            onOptOut={content.onOptOutAction}
+          />
+        </StyledCategoryActions>
+      </StyledCategoryContentDiv>
+    )
+  }
+
+  getAdContent = (content: AdContent, i: number) => {
+    return (
+      <StyledAdContentDiv>
+        <StyledAdLink href={content.brandUrl} target={'_blank'}>
+          {/* TODO(jsadler): Remove 'display: none' from StyledLogoDiv once icons are in place */}
+          <StyledLogoDiv>
+            <StyledLogo />
+          </StyledLogoDiv>
+          <StyledAdInfoDiv>
+            <StyledAdBrand>{content.brand}</StyledAdBrand>
+            <StyledAdInfo>{content.brandInfo}</StyledAdInfo>
+            <StyledAdInfo>{content.brandDisplayUrl}</StyledAdInfo>
+          </StyledAdInfoDiv>
+        </StyledAdLink>
+        <StyledAdStatDiv>
+          <StyledAdStat>
+            {content.adAction}
+          </StyledAdStat>
+          <div />
+          <StyledAdStatActions>
+            <ThumbLikePicker
+              likeAction={content.likeAction}
+              onThumbUpPress={content.onThumbUpPress}
+              onThumbDownPress={content.onThumbDownPress}
+            />
+            <DropMenu
+              onMenuSave={content.onMenuSave}
+              onMenuFlag={content.onMenuFlag}
+            />
+          </StyledAdStatActions>
+        </StyledAdStatDiv>
+      </StyledAdContentDiv>
+    )
+  }
+
+  getDetailRows = (rows: AdDetailRow[]): Row[] | undefined => {
+    if (!rows) {
+      return
+    }
+
+    return rows.map((row: AdDetailRow, i: number): Row => {
+      const cell: Row = {
+        content: [
+          {
+            content: (
+              <td key={i} />
+            ),
+            customStyle: {
+              width: '15% !important',
+              border: 'none'
+            }
+          },
+          {
+            content: (
+              this.getAdContent(row.adContent, i)
+            ),
+            customStyle: {
+              width: '70% !important',
```

Comment:
I know we have these inline styles in the `tableContribute` component, but could we offload these to styled components? Let me know if there was some other limitation

Reply:
Offloaded. In fact, only the td was effective.